src/content/en/fundamentals/media/_code/mobile-web-video-playback.js

```diff
@@ -0,0 +1,353 @@
+  videoControls.classList.remove('visible');
+}
+
+videoControls.addEventListener('click', function(event) {
+  if (videoControls.classList.contains('visible')) {
+    hideVideoControls();
+  } else {
+    showVideoControls();
+  }
+});
+
+document.addEventListener('DOMContentLoaded', function() {
+  showVideoControls();
+});
+
+
+/* Utils */
+
+function secondsToTimeCode(seconds) {
+  const h = Math.floor(seconds / 3600);
+  const m = Math.floor((seconds % 3600) / 60);
+  const s = Math.floor(seconds % 60);
+  return [
+    h,
+    m > 9 ? m : '0' + m,
+    s > 9 ? s : '0' + s,
+  ].filter(Boolean).join(':');
+}
+
+
+/* Custom controls */
+
+video.addEventListener('loadedmetadata', function() {
+  console.log('loadedmetadata');
+  videoDuration.textContent = secondsToTimeCode(video.duration);
+  videoCurrentTime.textContent = secondsToTimeCode(video.currentTime);
+  videoProgressBar.style.transform = `scaleX(${video.currentTime / video.duration})`;
+});
+
+playPauseButton.addEventListener('click', function(event) {
+  event.stopPropagation();
+  if (video.paused) {
+    video.play();
+  } else {
+    video.pause();
+  }
+});
+
+video.addEventListener('play', function() {
+  console.log('play');
+  playPauseButton.classList.add('paused');
+  hideVideoControls();
+});
+
+video.addEventListener('pause', function() {
+  console.log('pause');
+  playPauseButton.classList.remove('paused');
+  showVideoControls();
+});
+
+video.addEventListener('ended', function() {
+  console.log('ended');
+  playPauseButton.classList.remove('paused');
+  video.currentTime = 0;
+  showVideoControls();
+});
+
+document.addEventListener('fullscreenchange', function() {
+  toggleFullscreenButton.classList.toggle('active', document.fullscreenElement);
+});
+
+video.addEventListener('timeupdate', function() {
+  console.log('timeupdate');
+  if (!videoControls.classList.contains('visible')) {
+    return;
+  }
+  videoCurrentTime.textContent = secondsToTimeCode(video.currentTime);
+  videoProgressBar.style.transform = `scaleX(${video.currentTime / video.duration})`;
+});
+
+seekForwardButton.addEventListener('click', function(event) {
+  event.stopPropagation();
+  seekForward();
+});
+
+seekBackwardButton.addEventListener('click', function(event) {
+  event.stopPropagation();
+  seekBackward();
+});
+
+const skipTime = 10;
+
+function seekForward() {
+  video.currentTime = Math.min(video.currentTime + skipTime, video.duration);
+}
+
+function seekBackward() {
+  video.currentTime = Math.max(video.currentTime - skipTime, 0);
+}
+
+video.addEventListener('seeking', function() {
+  console.log('seeking');
+  video.classList.add('seeking');
+});
+
+video.addEventListener('seeked', function() {
+  console.log('seeked');
+  setTimeout(function() {
+    video.classList.remove('seeking');
+  }, 200);
+});
+
+
+/* Fullscreen */
+
+toggleFullscreenButton.addEventListener('click', function(event) {
+  event.stopPropagation();
+  if (document.fullscreenElement) {
```

Comment:
This will be undefined in all current browsers, so will always take the else branch. I don't have time to do a review of this section this week, but before publishing it would be good to run the code in desktop and mobile browsers based on the 4 different engines.